config.go

```diff
@@ -0,0 +1,526 @@
+// Copyright (C) 2015-2022 Lightning Labs and The Lightning Network Developers
+// Copyright (C) 2022 Bottlepay and The Lightning Network Developers
+
+package lndsigner
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	flags "github.com/jessevdk/go-flags"
+)
+
+const (
+	defaultConfigFilename  = "signer.conf"
+	defaultTLSCertFilename = "tls.cert"
+	defaultTLSKeyFilename  = "tls.key"
+	defaultLogLevel        = "info"
+	defaultLogDirname      = "logs"
+	defaultLogFilename     = "signer.log"
+	defaultRPCPort         = 10009
+	defaultRPCHost         = "localhost"
+
+	defaultMaxLogFiles    = 3
+	defaultMaxLogFileSize = 10
+
+	// DefaultAutogenValidity is the default validity of a self-signed
+	// certificate. The value corresponds to 14 months
+	// (14 months * 30 days * 24 hours).
+	defaultTLSCertDuration = 14 * 30 * 24 * time.Hour
+
+	// Set defaults for a health check which ensures that the TLS certificate
+	// is not expired. Although this check is off by default (not all setups
+	// require it), we still set the other default values so that the health
+	// check can be easily enabled with sane defaults.
+	defaultTLSInterval = time.Minute
+	defaultTLSTimeout  = time.Second * 5
+	defaultTLSBackoff  = time.Minute
+	defaultTLSAttempts = 0
+)
+
+var (
+	// DefaultSignerDir is the default directory where lnd tries to find its
+	// configuration file and store its data. This is a directory in the
+	// user's application data, for example:
+	//   C:\Users\<username>\AppData\Local\Lndsigner on Windows
+	//   ~/.lndsigner on Linux
+	//   ~/Library/Application Support/Lndsigner on MacOS
+	DefaultSignerDir = btcutil.AppDataDir("lndsigner", false)
+
+	// DefaultConfigFile is the default full path of lnd's configuration
+	// file.
+	DefaultConfigFile = filepath.Join(DefaultSignerDir, defaultConfigFilename)
+
+	defaultLogDir = filepath.Join(DefaultSignerDir, defaultLogDirname)
+
+	defaultTLSCertPath = filepath.Join(DefaultSignerDir, defaultTLSCertFilename)
+	defaultTLSKeyPath  = filepath.Join(DefaultSignerDir, defaultTLSKeyFilename)
+)
+
+// Config defines the configuration options for lnd.
+//
+// See LoadConfig for further details regarding the configuration
+// loading+parsing process.
+type Config struct {
+	SignerDir  string `long:"signerdir" description:"The base directory that contains signer's data, logs, configuration file, etc."`
+	ConfigFile string `short:"C" long:"configfile" description:"Path to configuration file"`
+
+	TLSCertPath        string        `long:"tlscertpath" description:"Path to write the TLS certificate for lnd's RPC services"`
+	TLSKeyPath         string        `long:"tlskeypath" description:"Path to write the TLS private key for lnd's RPC services"`
+	TLSExtraIPs        []string      `long:"tlsextraip" description:"Adds an extra ip to the generated certificate"`
```

Comment:
In the side-car model, this may be too much. Also the expiry could be indefinitely if we use it on localhost anyway?